rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -0,0 +1,202 @@
+    """Queries DB, returns results in format consumed by front-end
+
+    :param start_date: Start date object
+    :param end_date: End date object
+    :param filters: Objects representing filters specified in UI
+    :param stratification: How to stratify (layer) results, as in a stacked bar chart
+    :return: Filtered, stratified results by date
+    """
+
+    filters_sql = self.get_facets_sql(filters)
+
+    if str(stratification) == 'TOTAL':
+      strata = ['TOTAL']
+      sql = self.get_total_sql(filters_sql)
+    elif str(stratification) == 'ENROLLMENT_STATUS':
+      strata = [str(val) for val in EnrollmentStatus]
+      sql = self.get_enrollment_status_sql(filters_sql)
+    else:
+      raise BadRequest('Invalid stratification: %s' % stratification)
+
+    params = {'start_date': start_date, 'end_date': end_date}
+
+    results_by_date = []
+
+    with self.session() as session:
+      cursor = session.execute(sql, params)
+
+    # Iterate through each result (by date), transforming tabular SQL results
+    # into expected list-of-dictionaries response format
+    try:
+      results = cursor.fetchall()
+      for result in results:
+        date = result[-1]
+        metrics = {}
+        values = result[:-1]
+        for i, value in enumerate(values):
+          key = strata[i]
+          if value == None:
+            value = 0
+          metrics[key] = int(value)
+        results_by_date.append({
+          'date': str(date),
+          'metrics': metrics
+        })
+    finally:
+      cursor.close()
+
+    return results_by_date
+
+  def get_facets_sql(self, facets):
+    """Helper function to transform facets/filters selection into SQL
+
+    :param facets: Object representing facets and filters to apply to query results
+    :return: SQL for 'WHERE' clause, reflecting filters specified in UI
+    """
+
+    facets_sql = []
+
+    facet_map = {
+      'awardee_ids': 'hpo_id',
+      'enrollment_statuses': 'enrollment_status'
+    }
+
+    for facet in facets:
+      filters_sql = []
+      db_field = facet_map[facet]
+      filters = facets[facet]
+
+      # TODO:
+      # Consider using an IN clause with bound parameters, instead, which
+      # would be simpler than this,
+      #
+      # TODO:
+      # Consider using bound parameters here instead of inlining the values
+      # in the SQL. We do that in other places using this function:
+      #
+      # dao/database_utils.py#L16
+      #
+      # This may help the SQL perform slightly better since the execution
+      # plan for the query can be cached when the only thing changing are
+      # the bound params.
+      for q_filter in filters:
+        if str(q_filter) != '':
+          filters_sql.append('ps.' + db_field + ' = ' + str(int(q_filter)))
+      if len(filters_sql) > 0:
+        filters_sql = '(' + ' OR '.join(filters_sql) + ')'
+        facets_sql.append(filters_sql)
+
+    if len(facets_sql) > 0:
+      facets_sql = 'WHERE ' + ' AND '.join(facets_sql)
```

Comment:
I think we're always going to want a WHERE clause that filters on withdrawal status and (TEST awardee or example.com email address). See this for the latter:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/offline/metrics_export.py#L41

If you don't want to do that now, please file a JIRA issue tracking it and add a TODO.

Reply:
I've implemented filtering for the test awardee and email.  Withdrawal status filtering is somewhat different (and desirable), so I created DA-588 to track it.